### PR TITLE
Editorial suggestions from WGLC

### PR DESCRIPTION
**Pull request by tfpauly**

### CODE REVIEW

**Review by tireddy2 — APPROVED** [review]

File: draft-ietf-ipsecme-add-ike.xml [hunk 1]
```diff
@@ -391,36 +398,38 @@
...
-          ENCDNS_IP* attributes in the CFG_REQUEST. For each IP address family
-          the initiator MUST include exactly one attribute with the Length
-          field set to 0 if no specific DNS resolver is requested. The
-          initiator MAY include the ENCDNS_DIGEST_INFO attribute with a list
-          of hash algorithms that are supported by the encrypted DNS
-          client.</t>
+          <t>If the initiator supports encrypted DNS, it includes either or both
+          of the ENCDNS_IP4 and ENCDNS_IP6 attributes in its CFG_REQUEST. If
+          the initiator does not want to request specific DNS resolvers, it sets
+          the Length field to 0 for the attribute. The initiator MUST NOT send
```
Comment (smyslov) [inline draft-ietf-ipsecme-add-ike.xml:413]: I don't think this MUST NOT is necessary (at least in RFC 2119 language). If for some reason multiple empty attributes are sent, with the current language the responder has to treat this as a fatal error (violation of MUST NOT) and most probably tear down the connection. In fact it's harmless (except for waste of bandwidth).

**Review by smyslov — CHANGES_REQUESTED** [review]

File: draft-ietf-ipsecme-add-ike.xml (hunk 1 above)
Comment (smyslov) [inline draft-ietf-ipsecme-add-ike.xml]: Oops, I missed that the original text also has this requirement. I still feel there is an inconsistency. For example, the draft prohibits sending multiple empty attributes, but is silent about sending multiple attribute with the same suggested resolver. If we want to be fully consistent, then the requirement should be: the initiator MAY send several attributes, in which case all of them MUST be distinct (this implicitly includes empty attribute). The responder MUST ignore repeated attributes.

Comment (tireddy2) [reply draft-ietf-ipsecme-add-ike.xml]: The responder may typically ignore the violation but if the client tries to takes advantage of it to waste the cycles of the responder (e.g., repeating the attribute several hundred times; the resolver may want to treat it is a fatal error). we can probably say: 

If the receiver gets repeated attributes, it SHOULD ignore it.

Comment (smyslov) [reply draft-ietf-ipsecme-add-ike.xml]: I can live with SHOULD :-)

**Review by smyslov — CHANGES_REQUESTED** [review]

File: draft-ietf-ipsecme-add-ike.xml [hunk 2]
```diff
@@ -391,36 +398,38 @@
...
+          more than one empty attribute of a given type. The initiator MAY send
```
Comment (smyslov) [inline draft-ietf-ipsecme-add-ike.xml]:
```suggestion
          the Length field to 0 for the attribute. If the initiator sends multiple 
          attributes of a particular type in the request, all of them MUST be distinct (either be empty or containing different suggested resolvers). The responder SHOULD ignore repeated attributes if they are included in the request. The initiator MAY send
```

Comment (tireddy2) [reply draft-ietf-ipsecme-add-ike.xml]: Looks good.

File: draft-ietf-ipsecme-add-ike.xml [hunk 3]
```diff
@@ -133,26 +133,29 @@
 
     <abstract>
       <t>This document specifies new Internet Key Exchange Protocol Version 2
-      (IKEv2) Configuration Payload Attribute Types for encrypted DNS
-      protocols, such as DNS-over-HTTPS (DoH), DNS-over-TLS (DoT), and
-      DNS-over-QUIC (DoQ).</t>
+      (IKEv2) Configuration Payload Attribute Types to assign DNS resolvers
+      that support encrypted DNS protocols.</t>
```
Comment (boucadair) [inline draft-ietf-ipsecme-add-ike.xml]:
```suggestion
      that support encrypted DNS protocols, such as DNS-over-HTTPS (DoH), DNS-over-TLS (DoT), and
      DNS-over-QUIC (DoQ).</t>
```

**Pull request merged**